src/test/java/com/aws/greengrass/logmanager/model/LogFileGroupTest.java

```diff
@@ -47,4 +56,43 @@ void GIVEN_log_files_THEN_find_the_active_file() throws IOException, Interrupted
         assertFalse(logFileGroup.isActiveFile(file));
         assertTrue(logFileGroup.isActiveFile(file2));
     }
+
+    @Test
+    void GIVEN_logManagerGetsFiles_WHEN_fileRotateAfterRetrieved_THEN_itReturnsTheCorrectFiles() throws IOException {
+        // Given
+
+        File rotatedFile1 = createLogFileWithSize(directoryPath.resolve("test.log.1").toUri(), 1024 * 5);
+        File activeFile = createLogFileWithSize(directoryPath.resolve("test.log").toUri(), 1024);
+
+        Pattern filePattern = Pattern.compile("test.log\\w*", Pattern.CASE_INSENSITIVE);
+        Instant lastUploadedFileInstant = Instant.EPOCH;
+        LogFileGroup logGroup = new LogFileGroup(
+                new ArrayList<>(), filePattern, directoryPath.toUri(), lastUploadedFileInstant);
+
+        // When
+
+        File[] directoryFiles = directoryPath.toFile().listFiles();
+        rotateFilesByRenamingThem(activeFile, rotatedFile1);
+        List<LogFile> logFiles = logGroup.getLogFiles(directoryFiles);
+
+        // Assert
+
+        directoryFiles = directoryPath.toFile().listFiles();
+        assertNotNull(directoryFiles);
+        assertEquals(directoryFiles.length, logFiles.size());
+
+        Path[] expectedPaths = {
+            directoryPath.resolve("test.log.2"),
+            directoryPath.resolve("test.log.1"),
+            directoryPath.resolve("test.log")
+        };
+
+        List<Path> filePaths = logFiles.stream().map(File::toPath).collect(Collectors.toList());
+        for (int i = 0; i < expectedPaths.length; i++) {
```

Comment:
This check is temporary I will update once logfiles have the hard links to ensure we are getting the expected files